src/asl/synth.ts

```diff
@@ -1894,12 +1894,77 @@ export class ASL {
               return { value: !!argOutput.value, containsJsonPath: false };
             }
           });
+        } else if (ref === String) {
+          const [arg] = expr.args;
+          if (!arg) {
+            return {
+              // String()
+              value: "",
+              containsJsonPath: false,
+            };
+          }
+          return this.evalExprToJsonPathOrLiteral(arg.expr, (argOutput) => {
+            // String(var) or String({ key: var })
+            if (ASLGraph.isJsonPath(argOutput) || argOutput.containsJsonPath) {
+              const temp = this.newHeapVariable();
+              const jsonPathStates = this.normalizeOutputToJsonPath(argOutput);
+              const { jsonPath } = ASLGraph.isJsonPath(jsonPathStates)
+                ? jsonPathStates
+                : jsonPathStates.output;
+
+              return {
+                ...ASLGraph.joinSubStates(arg, jsonPathStates, {
+                  // if this was a literal, we know it was an object, not a string
```

Comment:
why?

Reply:
`containsJsonPath` is only possible for an Object literal

Reply:
>                   // if this was a literal, we know it is an object (only objects have containsJsonPath:true), not a string